Disher/View Controllers/RecipesViewController.m

```diff
@@ -303,48 +313,77 @@ - (void) queryAPIs:(NSString *) input withOption:(NSInteger) option completionHa
-//        }];
-//    });
+    dispatch_group_enter(group);
+    dispatch_group_async(group, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^ {
+        [self queryMealDB:input withOption:option completionHandler:^(NSArray *returnedMeals) {
+            NSString *recipeName;
+            NSString *imageLink;
+            NSString *mealID;
+            NSString *source;
+            NSArray<NSString *> *cuisine;
+            if (option == 0) {
+                for (NSDictionary *meal in returnedMeals) {
+                    recipeName = meal[@"strMeal"];
+                    imageLink = meal[@"strMealThumb"];
+                    mealID = meal[@"idMeal"];
+                    source = @"TheMealDB";
+                    cuisine = @[meal[@"strArea"]];
+                    [self.cuisines addObject:meal[@"strArea"]];
+                    Recipe *newRecipe = [Recipe initWithRecipe:recipeName withURL:imageLink withSource:source withID:mealID withCuisine:cuisine];
+                    [self.tableViewRecipes addObject:newRecipe];
+                    self.unfilteredTableViewRecipes = [self.tableViewRecipes copy];
+                    [self refreshData];
+                }
+            }
+            else if (option == 1) {
+                for (NSDictionary *meal in returnedMeals) {
+                    recipeName = meal[@"strMeal"];
+                    imageLink = meal[@"strMealThumb"];
+                    mealID = meal[@"idMeal"];
+                    source = @"TheMealDB";
+                    [Recipe getRecipeInfo:mealID withSource:@"TheMealDB" withCompletion:^(NSDictionary * _Nonnull recipeInformation) {
+                        [self.cuisines addObject:recipeInformation[@"strArea"]];
+                        self.temp = @[recipeInformation[@"strArea"]];
+                        NSArray *cuisine = self.temp;
+                        Recipe *newRecipe = [Recipe initWithRecipe:recipeName withURL:imageLink withSource:source withID:mealID withCuisine:cuisine];
+                        [self.tableViewRecipes addObject:newRecipe];
+                        self.unfilteredTableViewRecipes = [self.tableViewRecipes copy];
+                        NSArray *ownedIngredients = [input componentsSeparatedByString:@","];
+                        NSMutableArray *recipeIngredients = [NSMutableArray new];
+                        NSInteger i = 1;
+                        BOOL nullFound = NO;
+                        while (!nullFound) {
+                            NSString *ingredient = recipeInformation[[NSString stringWithFormat:@"strIngredient%ld", i]];
+                            if (ingredient != nil && ![ingredient isEqualToString:@""]) {
+                                [recipeIngredients addObject:ingredient];
+                                i++;
+                            }
+                            else {
+                                nullFound = YES;
+                            }
+                        }
```

Comment:
not really necessary but it could be helpful to include a link (as a comment on github) to what the format returned from the API looks like so the reviewer knows why you took this approach